### PR TITLE
Refactor user edit form

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9697

Updates:
1. Separates new user and edit user actions from `StaffForm` and adds more config props so the form uses what is handed to it
2. Componentizes the activate/inactivate button to isolate the activate/inactivate API calls
3. Componentizes the Staff form error modal to hopefully help debug https://github.com/cityofaustin/atd-data-tech/issues/10085
4. Componentizes the Staff form confirm modal

## Testing
**URL to test:** 
https://md-9697-user-edit-form--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Go to https://md-9697-refactor-user-edit-form--atd-moped-main.netlify.app/moped/staff, log in as an admin, and create a new user (I used `mike.dilley+editform@austintexas.gov`)
2. Edit your new user's details like role, first name, last name, title, workgroup, or password.
3. Inactivate your user
4. Activate your user with a new password (all fields besides password should be disabled)
5. Inactivate your user to cleanup the staging Cognito user pool and the staging DynamoDB table

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/staff/StaffForm.js [hunk 1]
```diff
@@ -128,83 +87,34 @@ const StaffForm = ({ editFormData = null, userCognitoId }) => {
...
 
   const {
     register,
     handleSubmit,
     errors,
     control,
     setValue,
-    getValues,
-    formState,
+    watch,
```
Comment (mddilley) [inline moped-editor/src/views/staff/StaffForm.js:97]: This update gives the confirm modal the latest and greatest values. Using `getValues` was occasionally returning `undefined` so the frontend password validation was failing even when a valid password was entered to reactivate a user.

**Comment by johnclary** [comment]
@mddilley when I login to this test branch the Hasura API is unreachable. I see for every query `POST https://md-9697-refactor-user-edit-form-graphql.moped-test.austinmobility.io/v1/graphql net::ERR_NAME_NOT_RESOLVED`. Might be an issue with the deployment?

File: moped-api/users/users.py [hunk 2]
```diff
@@ -227,16 +227,19 @@ def user_update_user(id: str, claims: list) -> (Response, int):
...
-            user_profile=request.json
+            user_profile=request.json, ignore_fields=ignore_fields
         )
 
         if not profile_valid:
             return jsonify({"error": profile_error_feedback}), 400
 
-        # If a password was included, let's reset it
-        if password is not None and password != "":
+        # If a password was included and not blank, let's reset it
```
Comment (johnclary) [inline moped-api/users/users.py:241]: 👍

**Comment by mddilley** [comment]
@johnclary I just created https://github.com/cityofaustin/atd-data-tech/issues/10273 to capture the DNS error that we have been seeing intermittently. Feel free to add any details - I just added the error you and I saw the other day, and the one you ran into here.

File: moped-editor/src/views/staff/helpers.js [hunk 3]
```diff
@@ -57,7 +72,49 @@ const errorsToTranslate = {
...
     : null;
+
+/**
+ * Makes sure the password looks ok
+ * @returns {boolean}
+ */
+export const passwordLooksGood = (password) =>
+  new RegExp(
+    "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$"
+  ).test(password);
```
Comment (johnclary) [inline moped-editor/src/views/staff/helpers.js:87]: ok, very out of scope, but, if we wanted to increase the min password length to, say, 12 chars, is that a matter of changing it here and the backend API? if it's minimal i think we should?

**Review by johnclary — APPROVED** [review]
wow, lots of really nice improvements here. this looks great and works great.

if it's easy enough to modify the Yup error messages, please do. 

🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

File: moped-editor/src/views/staff/EditStaffView.js [hunk 4]
```diff
@@ -11,90 +14,85 @@ import {
...
+const validationSchema = yup.object().shape({
+  first_name: yup.string().required(),
+  last_name: yup.string().required(),
+  title: yup.string().required(),
+  workgroup: yup.string().required(),
+  workgroup_id: yup.string().required(),
+  email: yup.string().required().email().lowercase(),
+  password: yup.string(),
+  roles: yup.string().required(),
+});
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-editor/src/views/staff/EditStaffView.js:30]: This is a great idea! I just added this to https://github.com/cityofaustin/atd-data-tech/issues/10085 to keep this PR from blowing up too much.

File: moped-editor/src/views/staff/NewStaffView.js [hunk 5]
```diff
@@ -8,26 +11,71 @@ import {
...
+const validationSchema = yup.object().shape({
+  first_name: yup.string().required(),
+  last_name: yup.string().required(),
+  title: yup.string().required(),
+  workgroup: yup.string().required(),
+  workgroup_id: yup.string().required(),
+  email: yup.string().required().email().lowercase(),
+  password: yup.string().required(),
+  roles: yup.string().required(),
+});
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-editor/src/views/staff/NewStaffView.js:37]: Same - added to https://github.com/cityofaustin/atd-data-tech/issues/10085

File: moped-editor/src/views/staff/helpers.js (hunk 3 above)
Comment (mddilley) [reply moped-editor/src/views/staff/helpers.js:87]: This would be a small change that would easily roll into this PR. I'll do this before merging. 👍

Comment (johnclary) [reply moped-editor/src/views/staff/helpers.js:87]: fantastic — thanks Mike!

**Review by chiaberry — APPROVED** [review]

**Pull request merged**